.buildkite/scripts/common.sh

```diff
@@ -56,8 +56,8 @@ with_msft_go() {
     create_workspace
     check_platform_architeture
     MSFT_DOWNLOAD_URL=https://aka.ms/golang/release/latest/go$(cat .go-version).${platform_type}-${arch_type}.tar.gz
-    retry 5 $(curl -sL -o - $MSFT_DOWNLOAD_URL | tar -xz -f - -C ${WORKSPACE})
-    export PATH="${PATH}:${WORKSPACE}/go/bin"
```

Comment:
This change is necessary because, prior to it, we were extracting `microsoft/go` inside the `fleet-server/bin` folder.  With that setup, running `go test ./...` from the `fleet-server` folder would result in an error like so:

```
panic: LoadImport called with empty package path
```

@v1v Are you OK with this change or do you have a better suggestion for where to extract `microsoft/go` and how to reference that location in these two lines?

Reply:
Shall we create a temporary folder and then point there to avoid any surprises?

```bash
    tempfolder=$(mktemp -d)
    retry 5 $(curl -sL -o - $MSFT_DOWNLOAD_URL | tar -xz -f - -C ${tempfolder}/)
    export PATH="${PATH}:${tempfolder}/go/bin"
```

Reply:
Done in 998d1d5b.